packages/server/src/sdk/app/projectApps/index.ts

```diff
@@ -0,0 +1,80 @@
+import { context, docIds, HTTPError, utils } from "@budibase/backend-core"
+import { DocumentType, ProjectApp, SEPARATOR } from "@budibase/types"
+
+async function guardName(name: string, id?: string) {
+  const existingProjectApps = await fetch()
+
+  if (existingProjectApps.find(p => p.name === name && p._id !== id)) {
+    throw new HTTPError(`App with name '${name}' is already taken.`, 400)
+  }
+}
+
+export async function fetch(): Promise<ProjectApp[]> {
+  const db = context.getAppDB()
+  const docs = await db.allDocs<ProjectApp>(
+    docIds.getProjectAppParams(null, { include_docs: true })
+  )
+  const result = docs.rows.map(r => ({
+    ...r.doc!,
+    _id: r.doc!._id!,
+    _rev: r.doc!._rev!,
+  }))
+  return result
+}
+
+export async function get(id: string): Promise<ProjectApp | undefined> {
+  const db = context.getAppDB()
+  const projectApp = await db.tryGet<ProjectApp>(id)
+  return projectApp
+}
+
+export async function create(
+  projectApp: Omit<ProjectApp, "_id" | "_rev" | "createdAt" | "updatedAt">
```

Comment:
Kinda feels like this pattern would be a nice utility type. `WithoutDocMetadata<ProjectApp>` or something.